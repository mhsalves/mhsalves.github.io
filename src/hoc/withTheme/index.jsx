import React from 'react';
import { ThemeProvider } from 'styled-components';
import themes from '../../bosons/themes';

const Wrapped = (Component) => (props) => (
  <ThemeProvider theme={themes.base}>
    <Component {...props} />
  </ThemeProvider>
);

export default Wrapped;
