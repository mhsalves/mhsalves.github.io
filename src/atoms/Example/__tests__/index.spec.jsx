import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import Example from '../index';
import withTheme from '../../../hoc/withTheme';

const WrappedExample = withTheme(Example);

describe('testing: Example component', () => {
  test('should render correctly', () => {
    const ExampleComponent = renderer.create(<WrappedExample />).toJSON();
    expect(ExampleComponent).toMatchSnapshot();
  });
  test('should has the example text', () => {
    const ExampleComponent = shallow(<Example />);
    expect(ExampleComponent.text()).toEqual('Example Component');
  });
});
