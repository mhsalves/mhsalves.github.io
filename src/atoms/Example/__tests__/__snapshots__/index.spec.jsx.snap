// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing: Example component should render correctly 1`] = `
.c0 {
  font-size: 1.333rem;
  line-height: 1.5rem;
  margin: 0;
  font-family: Nunito Sans,sans-serif;
  font-size: 1.2rem;
  line-height: 1.5rem;
}

@media (min-width:768px) {
  .c0 {
    font-size: 1.333rem;
    line-height: 1.5rem;
  }
}

@media (min-width:576px) {
  .c0 {
    margin: 24px auto;
    max-width: 1152px;
  }
}

<p
  className="c0"
>
  Example Component
</p>
`;
